lib/manager/cocoapods/readme.md

```diff
@@ -0,0 +1,232 @@
+- `>= 0.1` Version 0.1 and any higher version
+- `< 0.1` Any version lower than 0.1
+- `<= 0.1` Version 0.1 and any lower version
+
+In addition to the logic operators CocoaPods has an optimistic operator `~>`:
+
+- `~> 0.1.2` Version 0.1.2 and the versions up to 0.2, not including 0.2 and higher
+- `~> 0.1` Version 0.1 and the versions up to 1.0, not including 1.0 and higher
+- `~> 0` Version 0 and higher, this is basically the same as not having it.
+
+---
+
+#### Is this package manager used for applications, libraries, or both? If both, is there a way to tell which is which?
+
+Package manager used for applications only
+
+---
+
+#### If ranges are supported, are there any cases when Renovate should pin ranges to exact versions if rangeStrategy=auto?
+
+I don't see such reason.
+
+## Lookup
+
+#### Is a new datasource required? Provide details
+
+There is a index service for cocoapods. Developers always search for pod on this [site](https://github.com/CocoaPods):
+
+Also, all podspec are stored in one [repo](https://github.com/CocoaPods/Specs)
+
+---
+
+#### Will users need the capability to specify a custom host/registry to look up? Can it be found within the package files, or within other files inside the repository, or would it require Renovate configuration?
+
+User can provide remove git url ([details](https://guides.cocoapods.org/using/the-podfile.html#from-a-podspec-in-the-root-of-a-library-repo))
+
+`pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git'`
+
+---
+
+#### Do the package files contain any "constraints" on the parent language (e.g. supports only v3.x of Python) or platform (Linux, Windows, etc) that should be used in the lookup procedure?
+
+Cocoapods supports only mac platform.
+
+---
+
+#### Will users need the ability to configure language or other constraints using Renovate config?
+
+I would add an option to check dependencies with only specific swift version.
+
+## Artifacts
+
+#### Are lock files or checksum files used? Mandatory?
+
+Yes, lock file will be named `Podfile.lock`.
+
+---
+
+#### If so, what tool and exact commands should be used if updating 1 or more package versions in a dependency file?
+
+To update single package: `pod update SomePodName`
+
+To update all packages: `pod update`
+
+---
+
+#### If applicable, describe how the tool maintains a cache and if it can be controlled via CLI or env? Do you recommend the cache be kept or disabled/ignored?
+
+Cocoapods keeps cache in directory `${HOME}/Library/Caches/CocoaPods`
+
+You can get information about cache state and clean cache throutgh cocoapods ([details](https://guides.cocoapods.org/terminal/commands.html#group_cache))
+
+---
+
+#### If applicable, what command should be used to generate a lock file from scratch if you already have a package file? This will be used for "lock file maintenance".
+
+You can generate `Podfile.lock` by running `pod install`. This command also install required dependencies and link them to xcode project.
```

Comment:
On which platforms does "pod install" work? e.g. only OSX?

Reply:
`pod install` works on OS X. But this command fetches all dependencies and links them wot Xcode project. Obvious it's not what we need. Let me investigate this question deeper.

Reply:
I probably should expand the lock file questions in the default readme to fully capture what we need to do.

What we *want* to do is just update the lockfile. It's usually not a good idea to try to do this manually because most package managers do a lot of complex stuff with transitive dependencies. So that means we need a CLI command to run. Ideally every package manager would have a command that lets you update the lock file without needlessly downloading all the dependencies too, however most don't support that option so we end up just needing to eat the bandwidth and wasted time.

BTW it seems like you can run pod commands inside Docker/Linux as it can be installed using `gem`.

Reply:
We can definitely install cocoapods on linux machine through the gem, but I'm not sure that we will be able to run `pod install`, as this command requires to have xcode project i the same directory.

Reply:
Remember that we don't technically need to install anything - what we need is to be able to create and/or update a Podfile.lock, if given a Podfile.